addon/mixins/store.js

```diff
@@ -7,29 +7,50 @@ export default Mixin.create({
   modelFor: function(key) {
     let factory = this._super(key);
     if (factory._isPartialModel === true) {
-      this.generatePartialExtensionModel(key, factory);
-      this.generatePartialExtensionSerializer(key, factory);
+      this._generatePartialExtensionModel(key, factory);
+      this._generatePartialExtensionSerializer(key, factory);
     }
     return factory;
   },
 
-  createRecord: function(modelName /*, inputProperties */) {
-    let newRecord = this._super(...arguments);
-    let factory = newRecord.constructor;
+  createRecord: function(modelName, inputProperties) {
+    let factory = this.modelFor(modelName);
+    let relationshipsToBeAssigned = {};
+    let attributesFromPartialModels = [];
 
     if (factory._isPartialModel) {
       factory.eachRelationship((relationshipKey, descriptor) => {
```

Comment:
What about extracting `_partialDescriptors` to a new mixin for now, used by the `serializer` and the `store`, would make it more DRY already. `_extractExtensionsAttributes` from https://github.com/BookingSync/ember-data-partial-model/commit/f139046 seem to be a good candidate as well.